src/pspm_trim.m

```diff
@@ -195,17 +212,19 @@
         else
             try_num_end = str2double(marker_end_vals);
         end
-        try_num_end = str2double(marker_end_vals);
         if ~isempty(try_num_end)
-            g_endmarker = find(ndata{1}.markerinfo.value == try_num_end,1);
+            g_endmarker = find(data_struct.markerinfo.value == try_num_end,1);
         elseif ischar(marker_end_vals)
-            g_endmarker = find(strcmpi(ndata{1}.markerinfo.name,marker_end_vals),1);
+            g_endmarker = find(strcmpi(data_struct.markerinfo.name,marker_end_vals),1);
         else
             warning('ID:invalid_input', 'The value or name of the ending marker must be numeric or a string');
             return
         end
         % check if the markers are valid
-        if startmarker < 1 || g_endmarker < startmarker
+        if numel(startmarker) ~= 1 || ...
+                numel(g_endmarker) ~= 1 || ...
+                startmarker < 1 || ...
+                g_endmarker < startmarker
             warning('ID:invalid_input', 'No valid reference markers.\n'); return
```

Comment:
Think at line 237, the `ndata` can be changed to `data_struct` if preferred...

Reply:
changed